headers/utils-qt/retrying_future.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+#include <QObject>
+#include <QFuture>
+#include <QFutureWatcher>
+#include <QTimer>
+#include <memory>
+#include <type_traits>
+
+
+namespace Internal {
+
+template<typename Process, typename ReturnType>
+struct RetryFutureCtx
+{
+    RetryFutureCtx(const Process& process, const std::function<bool(const ReturnType&)>& checkFunction, unsigned int retryCount, unsigned int retryTimeout)
+        : futureProcess(process),
+          checker(checkFunction),
+          retryCnt(retryCount),
+          retryInterval(retryTimeout)
+    {
+        request();
+    }
+
+    ~RetryFutureCtx() {}
+
+    Process futureProcess;  // QFuture<T> func();
+    std::function<bool(ReturnType)> checker;
+
+    QObject tracker;
+    unsigned int retryCnt {0};
+    unsigned int retryInterval {0};
+    std::unique_ptr<QFutureWatcher<ReturnType>> watcher;
+    QFutureInterface<ReturnType> futureInterface;
+
+    void onFinished() {
+        retryCnt--;

```

Comment:
Seems will not work correctly if initial `retryCnt == 0`. Better move decrement to `onRetry` func, _there_ is ability to handle this case correctly.

Also, add test case for it, please.